java-generator/maven-plugin/src/main/java/io/fabric8/java/generator/maven/plugin/JavaGeneratorMojo.java

```diff
@@ -146,6 +146,9 @@ public void execute() throws MojoExecutionException {
           throw new MojoExecutionException("URL '" + url + "' is not valid", e);
         }
       }
+      if (!downloadTarget.isDirectory()) {
```

Comment:
the subfolder needs to be re-created in most cases

Reply:
Sorry for the late review, is this still necessary after the changes in the other PR  (#4815)